tools/data-api/internal/repositories/helpers.go

```diff
@@ -0,0 +1,53 @@
+package repositories
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+func listSubDirectories(path string) (*[]string, error) {
+	directories := make([]string, 0)
+
+	contents, err := os.ReadDir(path)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving list of sub directories under %q: %+v", path, err)
+	}
+
+	for _, c := range contents {
+		if c.IsDir() {
+			directories = append(directories, c.Name())
+		}
+	}
+
+	return &directories, nil
+}
+
+func loadJson(path string) (*[]byte, error) {
+	contents, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("loading %q", path)
+	}
+
+	defer contents.Close()
+
+	byteValue, err := io.ReadAll(contents)
+	if err != nil {
+		return nil, fmt.Errorf("reading contents of %q", path)
+	}
+
+	return &byteValue, nil
+}
+
+// getDefinitionInfo transforms the file names in the api definitions directory into a definition type and a name e.g.
+// Model-KeyVaultProperties.json -> type = Model and name = KeyVaultProperties
+func getDefinitionInfo(fileName string) (string, string) {
+	splitName := strings.Split(fileName, "-")
+
+	definitionType := splitName[0]
+	definitionName := strings.Split(splitName[1], ".")[0]
+
+	return definitionType, definitionName
+
+}
```

Comment:
This method assumes that we're only parsing JSON files - so `.DS_Store` files and similar (output by macOS Finder/Windows Explorer) would crash this, it's probably worth returning an error here if the file extension isn't `*.json` since that would indicate we're loading files we're not intending too?